demo/2D/Karman_vortex_street.py

```diff
@@ -1,142 +1,273 @@
 """
-from six.moves import range
+
 import numpy as np
 import sympy as sp
-
 import pylbm
 
-X, Y, LA = sp.symbols('X, Y, lambda')
-rho, qx, qy = sp.symbols('rho, qx, qy')
+# pylint: disable=redefined-outer-name
+
+X, Y = sp.symbols('X, Y')
+RHO, QX, QY = sp.symbols('rho, qx, qy')
+LA = sp.symbols('lambda', constants=True)
+
+
+# pylint: disable=unused-argument
+def bc_in(f, m, x, y, rho_o, u_o):
+    """
+    boundary values on the left bound
+    """
+    m[RHO] = rho_o
+    m[QX] = rho_o*u_o
+    m[QY] = 0.
 
-def bc_rect(f, m, x, y, rhoo, uo):
-    m[rho] = 0.
-    m[qx] = rhoo*uo
-    m[qy] = 0.
 
 def vorticity(sol):
-    qx_n = sol.m[qx]
-    qy_n = sol.m[qy]
-    vort = np.abs(qx_n[1:-1, 2:] - qx_n[1:-1, :-2]
-                  - qy_n[2:, 1:-1] + qy_n[:-2, 1:-1])
-    return vort.T
+    """
+    compute the vorticity of the solution
+    """
+    qx_n = sol.m[QX]
+    qy_n = sol.m[QY]
+    vort = np.abs(
+        qx_n[1:-1, 2:] - qx_n[1:-1, :-2] -
+        qy_n[2:, 1:-1] + qy_n[:-2, 1:-1]
+    )
+    return vort
 
-def run(dx, Tf, generator="cython", sorder=None, withPlot=True):
+
+# pylint: disable=invalid-name
+def run(space_step,
+        final_time,
+        generator="cython",
+        sorder=None,
+        with_plot=True):
     """
     Parameters
     ----------
 
-    dx: double
+    space_step: double
         spatial step
 
-    Tf: double
+    final_time: double
         final time
 
-    generator: pylbm generator
+    generator: string
+        pylbm generator
 
     sorder: list
         storage order
 
-    withPlot: boolean
+    with_plot: boolean
         if True plot the solution otherwise just compute the solution
 
+
+    Returns
+    -------
+
+    sol
+        <class 'pylbm.simulation.Simulation'>
+
     """
     # parameters
-    xmin, xmax, ymin, ymax = 0., 2., 0., 1.
-    radius = 0.125
-    la = 1. # velocity of the scheme
-    rhoo = 1.
-    uo = 0.05
-    mu   = 2.5e-5 #0.00185
-    zeta = 10*mu
-    dummy = 3.0/(la*rhoo*dx)
-    s1 = 1.0/(0.5+zeta*dummy)
-    s2 = 1.0/(0.5+mu*dummy)
-    s  = [0.,0.,0.,s1,s1,s1,s1,s2,s2]
-    dummy = 1./(LA**2*rhoo)
-    qx2 = dummy*qx**2
-    qy2 = dummy*qy**2
-    q2  = qx2+qy2
-    qxy = dummy*qx*qy
-
-    dico = {
-        'box':{'x':[xmin, xmax], 'y':[ymin, ymax], 'label':[0, 1, 0, 0]},
-        'elements':[pylbm.Circle([.3, 0.5*(ymin+ymax)+2*dx], radius, label=2)],
-        'space_step':dx,
+    scheme_name = 'Geier'
+    xmin, xmax, ymin, ymax = 0., 2., 0., 1.  # bounds of the domain
+    radius = 1./32                           # radius of the obstacle
+    la = 1.                                  # velocity of the scheme
+    rho_o = 1.                               # reference value of the mass
+    u_o = 0.05                               # boundary value of the velocity
+    mu = 5.e-6                               # bulk viscosity
+    zeta = 10*mu                             # shear viscosity
+
+    def moments_choice(scheme_name, mu, zeta):
```

Comment:
This function is found in various examples.
I think we can extract it in a separate python file and import it.